### PR TITLE
Adds spacy entity extraction as an option

### PR DESCRIPTION
**Pull request by MaxReinisch**
Also now requires a `config.json` file.  I am not sure if I did this part correctly.

### CODE REVIEW

File: tvnews/NewsRecommender.py [hunk 1]
```diff
@@ -22,26 +25,50 @@
...
 http = urllib3.PoolManager()
+nlp = en_core_web_sm.load()
 
-def makeRecommendations(article):
-    calais_json = getOpenCalaisResponse(article)
-    entities = getSearchQuery(calais_json)
-    GDELT_response = getGDELTv2Response(entities) #GDELT_response in JSON format
+with open('config.json') as json_data_file:
+    config = json.load(json_data_file)
+    print(config)
```
Comment (vbanos) [inline tvnews/NewsRecommender.py]: Forgot a `print` here.

**Comment by vbanos** [comment]
You need to remove the `config.json` file requirement. Please let me explain why:

`tvnews` is a library. The user will use it along with N other libraries in a program.
If each library requires its own config file, the final program will have N different config
files and it will be a configuration nightmare.
If `tvnews` was a stand-alone program, using a conf file would be the correct approach.

Make your relevant functions receive the required options e.g.
```
getOpenCalaisResponse(article, calais_endpoint, calais_header, ...)
```
Its no problem to have a function with a couple of arguments.

Then, the user of the library will add the `tvnews` conf options in his general app conf file and will
pass the relevant options to your methods when using them.

**Comment by vbanos** [comment]
There is also another serious problem with your implementation (I mention this just for learning purposes).
You have the code opening the `config.json` sitting bare inside the python module. Its not in any function.
So, when the user imports you'll module, the code will try to load a conf file without calling any function.
That's very unexpected and will create confusion and problems to the user.
Always have a `load_conf` type of function to do this.

In any case, you won't need that now as you don't need the `config.json` file at all.

**Comment by MaxReinisch** [comment]
Thanks for the config lessons.  I'm always happy to learn.  

I am going to fix this and commit to this PR.  I also think that it will be useful to default to spacy if opencalais fails

**Comment by vbanos** [comment]
Looks good, please merge and I'll try to use it ASAP.

**Pull request merged**